src/Halogen/VDom/DOM/Prop.purs

```diff
@@ -0,0 +1,238 @@
+module Halogen.VDom.DOM.Prop
+  ( Prop(..)
+  , Namespace(..)
+  , ElemRef(..)
+  , PropValue
+  , propFromString
+  , propFromBoolean
+  , propFromInt
+  , propFromNumber
+  , buildProp
+  ) where
+
+import Prelude
+import Control.Monad.Eff (Eff)
+import Control.Monad.Eff.Ref (REF)
+import Control.Monad.Eff.Ref as Ref
+import Data.Maybe (Maybe(..), maybe)
+import Data.StrMap (StrMap)
+import Data.StrMap as StrMap
+import Data.StrMap.ST (STStrMap)
+import Data.StrMap.ST as STStrMap
+import Data.Nullable (Nullable, toNullable)
+import Data.Function.Uncurried as Fn
+import Data.Newtype (class Newtype)
+import Data.Tuple (Tuple(..), fst, snd)
+import DOM (DOM)
+import DOM.Event.EventTarget (EventListener, eventListener) as DOM
+import DOM.Event.Types (EventType(..), Event) as DOM
+import DOM.HTML.Types (HTMLElement) as DOM
+import DOM.Node.Types (Element) as DOM
+import Halogen.VDom as V
+import Halogen.VDom.Util (refEq, diffWithKeyAndIxE, strMapWithIxE)
+import Unsafe.Coerce (unsafeCoerce)
+
+data Prop a
+  = Attribute (Maybe Namespace) String String
+  | Property String PropValue
+  | Handler DOM.EventType (DOM.Event → Maybe a)
+  | Ref (ElemRef DOM.HTMLElement → Maybe a)
```

Comment:
I added this instead of a `Maybe` because I felt it was more useful/informative.